app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -78,13 +73,25 @@ def scoped_search(query, default_sort_by, default_sort_order, options = {})
       page = params[:page] || 1
       per_page = params[:per_page] || ::Setting::General.entries_per_page
 
+      query = (total.zero? || sub_total.zero?) ? [] : query
```

Comment:
when would `sub_total` be zero, but `total` nonzero? If someone selected off the end of the the pagination in the API call?